coqpyt/tests/proof_file/utility.py

```diff
@@ -41,6 +42,11 @@ def setup(self, file_path, workspace=None):
         self.proof_file.run()
         self.versionId = VersionedTextDocumentIdentifier(uri, 1)
 
+        output = subprocess.check_output(f"coqtop -v", shell=True)
+        coq_version = output.decode("utf-8").split("\n")[0].split()[-1]
+        default = version.parse(coq_version) < version.parse("8.19")
+        self.coq_version = "default" if default else "8.19"
```

Comment:
I understand what you did here... because this is basically what we talked about, but now that I think about it, it is not the best approach. The `self.coq_version` should just be the `coq_version`. Then what happens when checking is:
1. If there is a match with a specific version, that version is used.
2. If there isn't, the default defined in the YAML file is used.

We just have to change a bit the code in `get_test_proofs` for this to work. Check if `coq_version` is in `x`. If it isn't, it checks if `default` is in `x`. Else, returns `x`.

Reply:
Alright, makes sense. Fixed in the latest commit.